src/robot/conf/settings.py

```diff
@@ -133,6 +134,12 @@ def _process_value(self, name, value):
             self._validate_flatten_keywords(value)
         return value
 
+    def _process_listener_value(self, value):
+        if isinstance(value, basestring):
+            return self._split_args_from_name_or_path(value)
+        else:
+            return (value, [])
```

Comment:
I understand that you only wanted to support listeners, but I nevertheless think it would be OK to just modify `_split_args_from_name_or_path` to have this at the beginning:

```python
if not isinstance(value, basestring):
    return value, []
```

If someone passes non-string value to other settings, things will break anyway. A big benefit would be that I could add support for instances to model modifiers separately.